File: Cpp/1122_Relative_Sort_Array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        vector<int> result;
        map<int, int> frequency;
        
        for(int i=0; i<arr1.size(); ++i){
            ++frequency[arr1[i]];
        }
        
        for(int i=0; i<arr2.size(); ++i){
            for(int j=0; j<frequency[arr2[i]]; ++j){
                result.push_back(arr2[i]);
            }
            frequency.erase(arr2[i]);
        }
        
        for(auto num : frequency){
            for(int i=0; i<num.second; ++i){
                result.push_back(num.first);
            }
        }
        
        return result;
    }
};

File: Kotlin/1122_Relative_Sort_Array.kt
class Solution {
    fun relativeSortArray(arr1: IntArray, arr2: IntArray): IntArray {
        val result = mutableListOf<Int>()
        val frequency = TreeMap<Int, Int>()
        
        arr1.forEach{
            frequency.put(it, frequency.getOrDefault(it, 0) + 1)
        }

        arr2.forEach{
            for(i in 0 until frequency[it]!!){
                result.add(it)
            }

            frequency[it] = 0
        }
        
        frequency.forEach{ (key, value) ->
            for(i in 0 until value){
                result.add(key)
            }
        }
        
        return result.toIntArray()
    }
}